crates/nu-command/src/filters/drop/drop_.rs

```diff
@@ -76,30 +76,23 @@ impl Command for Drop {
         call: &Call,
         input: PipelineData,
     ) -> Result<PipelineData, ShellError> {
+        let head = call.head;
         let metadata = input.metadata();
-        let rows: Option<i64> = call.opt(engine_state, stack, 0)?;
-        let v: Vec<_> = input.into_iter_strict(call.head)?.collect();
-        let vlen: i64 = v.len() as i64;
+        let rows: Option<Spanned<i64>> = call.opt(engine_state, stack, 0)?;
+        let mut values = input.into_iter_strict(head)?.collect::<Vec<_>>();
 
-        let rows_to_drop = if let Some(quantity) = rows {
-            quantity
+        let rows_to_drop = if let Some(rows) = rows {
+            if rows.item < 0 {
+                return Err(ShellError::NeedsPositiveValue { span: rows.span });
```

Comment:
Is this a breaking change?

Reply:
Oops, yes it its. Updated the description.